Exercise1_Test.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+
+import Exercise1
+import unittest
+
+class Exercise1Test(unittest.TestCase):
+
+  def testBaseCases(self):
+    self.assertEquals(True, Exercise1.is_permutation("", ""))
+    self.assertEquals(False, Exercise1.is_permutation(None, None))
+
+  def testKnownCases(self):
```

Comment:
Since `sys.maxsize` is very very large (`9223372036854775807` for my installation of Python) a string with that many characters would require 9 exabytes of memory (9 billion gigabytes). The tests are probably taking a long time as they require more and more memory (and you are actually each time allocating a new strings with yet one more character in it). So this is not really feasible to test.

I would give up on such a test: you can use a long string, but `sys.maxsize` is just too much.

For the case of a long string, you might also want to consider the following:

```python
>>> "a" * 10 == "aaaaaaaaaa"
True
```